ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/config/AuditYamlConfig.java

```diff
@@ -48,16 +56,35 @@ boolean isFromFile()
      *
      * @return the list of whitelisted users
      */
-    public List<String> getWhitelist()
+    List<String> getWhitelist()
     {
         return whitelist != null ? Collections.unmodifiableList(whitelist) : DEFAULT_WHITELIST;
     }
 
-    /**
-     * @return the SLF4J configuration or {@code null} if not configured.
-     */
-    public AuditYamlSlf4jConfig getSlf4j()
+    ParameterizedClass getLoggerBackendParameters()
     {
-        return slf4j;
+        // We hand out a deep copy since:
+        // - ParameterizedClass is mutable
+        // - The original map of options may contain a mix of Integers and Strings
+        //   Converting it all to Strings makes life easier for plug-ins when parsing options
+        return logger_backend != null ? deepToStringCopy(logger_backend) : deepToStringCopy(DEFAULT_LOGGER_BACKEND);
```

Comment:
TED: Why is all this (deep copy) needed? No need to deep clone a string->string map? Should be sufficient to just copy the map...
return ParameterizedClass(original.class_name, new HashMap<>(original.parameters));
or just use Collections.unmodifiableMap(original.parameters

Reply:
Deep copy with toString conversion necessary since Map values can be a mix of String and Integer. Added comments to clarify.